.github/workflows/lint.yml

```diff
@@ -16,10 +16,15 @@ jobs:
           python-version: 3.8
 
       - name: Setup Poetry
-        uses: Gr1N/setup-poetry@v4
+        run: |
+          export POETRY_HOME=/opt/poetry
+          python3 install-poetry.py --version 1.2.0
```

Comment:
Is this install-poetry.py the same one found here?

https://github.com/python-poetry/install.python-poetry.org/blob/main/install-poetry.py

The [Poetry CI recommendations](https://python-poetry.org/docs/#ci-recommendations) support what you have done here:

> Downloading a copy of the installer script to a place accessible by your CI pipelines (or maintaining a copy of the [repository](https://github.com/python-poetry/install.python-poetry.org)) is strongly suggested, to ensure your pipeline’s stability and to maintain control over what code is executed.

But it seems to me quite a big third-party dependency to include in the repo just for installing Poetry.

The same document also recommends using pip in virtual environment:

```bash
export POETRY_HOME=/opt/poetry
python3 -m venv $POETRY_HOME
$POETRY_HOME/bin/pip install poetry==1.2.0
$POETRY_HOME/bin/poetry --version
```

I think the pip method is a lot cleaner, if it works. But I haven't tried it in Github Actions yet.

Did you consider this solution?

Reply:
https://github.com/connelldave/botocove/runs/8172235417?check_suite_focus=true

This is the reason I removed the action, I was less interested in the clean way and more that CI would just work on my other PR. I just went with the guidance that didn't involve me curling the script directly each run.

I don't think there's any real relevance to adding a different build dependency that's only used in a GitHub action, and not relying on pypi for the packaging dependency seems reasonable enough to not worry about this very much 🤷

Reply:
So the old way just started failing. Oh dear! That's a good reason to change it :-)

```
Version 1.2.0 is not supported by this installer! Please specify a version prior to 1.2.0a1 to continue!
Error: The process '/opt/hostedtoolcache/Python/3.8.13/x64/bin/python' failed with exit code 1
```

> not relying on pypi for the packaging dependency

Not sure what you mean by this part. I thought the packaging part necessarily depended on PyPI because that is where the package is uploaded to.

I agree that it's nothing to worry about. I was just surprised to see the change but that build log makes it pretty clear why it's necessary!

Reply:
> Not sure what you mean by this part. I thought the packaging part necessarily depended on PyPI because that is where the package is uploaded to.

My vague thinking is that it feels better not to have a circular dependency in any system - I don't want to pip install pip, so to speak. Mostly philosoposphical irrelevance, it'd be fine to just pip install poetry too but given there's an alternative I preferred it.